### ISSUE TITLE
Differences in botskills --luisfolder parameter definition and usage[Docs]

### ISSUE THREAD

**Issue by saikumaru** [microsoft/botframework-solutions#3489]
/skills/handbook/add-skills-to-a-virtual-assistant/

For "botskills connect" in the documentation it is mentioned as

>The --luisFolder parameter can be used to point the Skill CLI at the source LU files for trigger utterances. For Skills provided within this repo these can be found in the **Deployment/Resources/LU folder of each Skill.**

In the command definition it is mentioned as 

>--luisFolder [path] | (OPTIONAL) Path to the folder containing your Skills' '.lu' files (defaults to '.**/deployment/resources/skills/en' inside your Virtual Assistant folder**)

Please can we have some clarity on this? 
Is it expected to have the skill's LU files copied over to this folder of the VA?

Or, should the command be pointing to the LU folder of the skill?

**Comment by Batta32** [comment]
Thanks @saikumaru for reporting this issue. We will be reviewing this and we will back to you ASAP!

**Comment by saikumaru** [comment]
@Batta32, do you have an update on this please?
Because it is sufficiently confusing as to where the LU files need to be.
More specifically,
(a)If we are trying to build the dispatch for a remote skill
(b)if we are trying to build the dispatch for a local skill

**Comment by Batta32** [comment]
Hi @saikumaru, sorry for the delay. The `--luisFolder` parameter **defines** the folder where the `Skill's .lu` files will be taken.

[Botskills](https://microsoft.github.io/botframework-solutions/skills/handbook/botskills/#commands) assumes that you are running the CLI within the Virtual Assistant project directory and have created your Bot through the template, where the LU files are inside the [Skills](https://github.com/microsoft/botframework-solutions/tree/master/samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/Deployment/Resources/Skills) folder.

>Is it expected to have the skill's LU files copied over to this folder of the VA?
Or, should the command be pointing to the LU folder of the skill?

If you don't have the Skill's .lu files in the Skills folder, you should define the location where you have the Skill's .lu files.

> Because it is sufficiently confusing as to where the LU files need to be.
More specifically,
(a)If we are trying to build the dispatch for a remote skill
(b)if we are trying to build the dispatch for a local skill

Example having the .lu files inside a local skill (specifically in the `deployment\resources\LU` folder of the Skill):

`botskills connect --remoteManifest <MANIFEST> --luisFolder "<PATH_TO_SKILL_DEPLOYMENT/RESOURCES/LU_FOLDER>"`

Hope this helps to you. As **next step**, we will be reviewing the inconsistencies in the botskills documentation to fix them, as soon as we have any update we will back to you 😊.

**Comment by Batta32** [comment]
Hi @saikumaru , we created the PR #3567 which solves this issue!
Feel free to add the feedback in the PR.

**Comment by saikumaru** [comment]
Thank you for clarifying that, really appreciate it. I have also added a comment on the PR.

Now that I am clear on the folder structure, can you re look at how your sample is built?
There are folders named with SKILL_NAME instead there should be folders of CULTUREs inside it. It was one of the initial confusions I had.

Folders under
https://github.com/microsoft/botframework-solutions/tree/master/samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/Deployment/Resources/Skills

**Comment by Batta32** [comment]
Thanks @saikumaru! We will be reviewing this too!

**Comment by Batta32** [comment]
Hi @saikumaru - sorry for the delay. As all the Skills are in the [botframework-skills](https://github.com/microsoft/botframework-skills) repository, we have references to them inside the Virtual Assistant simplifying the use of Botskills for the connection of those Skills.

In the case of TypeScript we have that implemented in [1.0 GA release](https://github.com/microsoft/botframework-solutions/pull/3584), and for C# you can find it in [master](https://github.com/southworks/botframework-solutions/tree/master/samples/csharp/assistants/virtual-assistant/VirtualAssistantSample/Deployment/Resources/Skills) branch.

However, we will be reviewing this, checking if it's necessary to add the remaining cultures for the Skills inside the Virtual Assistant.

We will back as soon as we have any update 😊.

**Comment by saikumaru** [comment]
Sure, would be glad to see the updates.

What I meant to say is, currently the folder structure is:

```
Resources/Skills/bingsearchskill/en-us
Resources/Skills/calendarskill/en-us
Resources/Skills/emailskill/en-us
```

etc

But it should me more like

```
Resources/Skills/en-us/bingsearchskill
Resources/Skills/en-us/calendarskill
Resources/Skills/en-us/emailskill
```

Because thats how your botskills command is configured to pick the .lu files.

**Comment by Batta32** [comment]
@saikumaru - we confirmed what you said and we noticed that the `--luisFolder` is pointing to the `deployment/resources/skills` and the `culture` is appended on it forming the path you mentioned: `deployment/resources/skills/en-us/`.

We will be reviewing which will be the best approach in order to fix this. We will back as soon as we have any update 😊.

![image](https://user-images.githubusercontent.com/11904023/89552517-9fb15080-d7e2-11ea-98f3-70a251e14f7d.png)

**Comment by Batta32** [comment]
Hi @saikumaru - After researching about the Skills folder, we concluded that those LU files are used in the deployment resources to **support our pipelines** that allows us to test connecting skills to the VA. If we change the structure it could break the tests.

We will follow up how to keep the test pipelines running without needing to check in the skills LU files underneath the deployment resources folder in the Virtual Assistant Sample project.

Just to clarify, you can use the Skills LU files from the [botframework-Skills](https://github.com/microsoft/botframework-skills/tree/master/skills/csharp) repository to connect the Skill to the Virtual Assistant.

As next step, we will clarify the documentation if it's necessary changing or adding the right guidance to avoid issues related to the Skills folder structure.

We will back as soon as we have any update 😊.

**Comment by Batta32** [comment]
@saikumaru - we recently updated the PR #3567 with this [commit](https://github.com/microsoft/botframework-solutions/pull/3567/commits/d8539db3de035932987285f52ec8347c728e327f).

If you have any feedback, don't hesitate to indicate it in the PR.

**Comment by Batta32** [comment]
This issue is solved with PR #3567.
